src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -185,7 +180,138 @@ export const WorkspaceMenu = ({ coreStart }: Props) => {
+                  }}
+                >
+                  <FormattedMessage
+                    id="core.ui.primaryNav.workspace.manage"
+                    defaultMessage="Manage workspace"
+                  />
+                </EuiButton>
+              </EuiFlexItem>
+            </>
+          ) : (
+            <>
+              <EuiFlexItem grow={false}>
+                <EuiAvatar size="m" color="plain" name="spacesApp" iconType="spacesApp" />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false} data-test-subj="context-menu-current-workspace-name">
+                {currentWorkspaceName}
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButton
+                  color="text"
+                  onClick={() => {
+                    window.location.assign(
+                      cleanWorkspaceId(
+                        coreStart.application.getUrlForApp(WORKSPACE_LIST_APP_ID, {
+                          absolute: false,
+                        })
+                      )
+                    );
+                  }}
+                >
+                  <FormattedMessage
+                    id="core.ui.primaryNav.workspaces.manage"
+                    defaultMessage="Manage workspaces"
+                  />
+                </EuiButton>
+              </EuiFlexItem>
+            </>
+          )}
+        </EuiFlexGroup>
+      </EuiPanel>
+      <EuiContextMenu
+        initialPanelId={0}
+        panels={recentWorkspacesPanels}
+        size="s"
+        data-test-subj="context-menu-recent-workspaces"
+      />
+      <EuiContextMenu
+        initialPanelId={0}
+        panels={allWorkspacesPanels}
+        size="s"
+        data-test-subj="context-menu-all-workspaces"
+      />
+      <EuiPanel paddingSize="s" hasBorder={false} color="transparent">
+        <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" gutterSize="s">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              size="xs"
+              flush="left"
+              key={WORKSPACE_LIST_APP_ID}
+              data-test-subj="context-menu-view-all-button"
+              onClick={() => {
+                window.location.assign(
+                  cleanWorkspaceId(
+                    coreStart.application.getUrlForApp(WORKSPACE_LIST_APP_ID, {
+                      absolute: false,
+                    })
+                  )
+                );
+              }}
+            >
+              <FormattedMessage id="core.ui.primaryNav.allWorkspace" defaultMessage="View all" />
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          {isDashboardAdmin && (
+            <EuiFlexItem grow={false}>
+              <EuiButton
+                color="text"
+                iconType="plus"
+                size="s"
+                key={WORKSPACE_CREATE_APP_ID}
+                data-test-subj="context-menu-create-workspace-button"
+                onClick={() => {
```

Comment:
Nit: Shall we need to check if it's in workspace? I think we don't need page refresh if out of workspace. The user experience would be more smooth.

Reply:
You mean i should delete `cleanWorkspaceId()`?

Reply:
That's not a big concerns. I'm prefer using `application.navigateToApp` when out of workspace. It won't cause whole page refreshing.

Reply:
That's not a big concerns. I'm prefer using `application.navigateToApp` when out of workspace. It won't cause whole page refreshing.

Reply:
Make sense, I will use application.navigateToApp for create/list workspace page.